File: impl/app/android/app/src/test/java/com/ak/contexta/ui/reading/TranslationModeTest.kt
package com.ak.contexta.ui.reading

import org.junit.Assert.assertEquals
import org.junit.Test

class TranslationModeTest {

    @Test
    fun `cycle order is FULL to DIM to BLURRED to HIDDEN back to FULL`() {
        assertEquals(TranslationMode.DIM, TranslationMode.FULL.next)
        assertEquals(TranslationMode.BLURRED, TranslationMode.DIM.next)
        assertEquals(TranslationMode.HIDDEN, TranslationMode.BLURRED.next)
        assertEquals(TranslationMode.FULL, TranslationMode.HIDDEN.next)
    }
}

File: impl/app/flutter/test/ui/reading/translation_mode_test.dart
import 'package:contexta/ui/reading/translation_visibility.dart';
import 'package:flutter_test/flutter_test.dart';

/// 译文模式循环测试（移植 TranslationModeTest.kt）。
void main() {
  test('cycle order is FULL to DIM to BLURRED to HIDDEN back to FULL', () {
    expect(TranslationMode.full.next, TranslationMode.dim);
    expect(TranslationMode.dim.next, TranslationMode.blurred);
    expect(TranslationMode.blurred.next, TranslationMode.hidden);
    expect(TranslationMode.hidden.next, TranslationMode.full);
  });
}
